Dockerfile

```
@@ -0,0 +1,19 @@
+FROM python:3.11-buster as base
+
+RUN apt-get update
+RUN apt-get install -y curl
+RUN curl -sSL https://install.python-poetry.org | POETRY_HOME=/opt/poetry python3 -
+ENV PATH=/opt/poetry/bin:$PATH
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-root
+COPY . /opt/todoapp/
+WORKDIR /opt/todoapp
+ENV WEBAPP_PORT=8000
+EXPOSE ${WEBAPP_PORT}
+RUN poetry install --without dev
+
+FROM base as development
+CMD ["poetry", "run", "flask", "run", "--host", "0.0.0.0"]
+
+FROM base as production
+CMD ["poetry", "run", "gunicorn", "--bind", "0.0.0.0", "todo_app.app:create_app()"]
```

Here you specify the WEBAPP_PORT but then only use it for one of your two stages, and similar you install the non-dev dependencies, but that feels like it might miss something in your dev container!

Where appropriate, don't be afraid to duplicate/override for the derived stages:
```suggestion
RUN poetry install --without dev

FROM base as development
ENV WEBAPP_PORT=5000
EXPOSE ${WEBAPP_PORT}
RUN poetry install
CMD ["poetry", "run", "flask", "run", "--host", "0.0.0.0"]

FROM base as production
ENV WEBAPP_PORT=8000
EXPOSE ${WEBAPP_PORT}

CMD ["poetry", "run", "gunicorn", "--bind", "0.0.0.0", "todo_app.app:create_app()"]
```
(This approach lets us share the layer that installs most of the packages, but adds any missing ones in the dev layer explicitly. Note that we haven't specified any dev-only packages yet, so this is admittedly overkill!)